packages/deployed-backend-client/src/backend_output_client.ts

```diff
@@ -22,6 +27,64 @@ export class DefaultBackendOutputClient implements BackendOutputClient {
       this.amplifyClient
     ).getStrategy(backendIdentifier);
     const output = await outputFetcher.fetchBackendOutput();
-    return unifiedBackendOutputSchema.parse(output);
+
+    return unifiedBackendOutputSchema.parse({
+      ...output,
+      ...(output[storageOutputKey] && {
+        [storageOutputKey]: this.parseStorageOutput(output[storageOutputKey]),
+      }),
+    });
   };
+
+  /**
+   * Parses the storage output and returns the modified backend output.
+   * @param storageOutput The storage output to parse.
+   * @returns The modified backend output.
+   */
+  private parseStorageOutput(
+    storageOutput: BackendOutputEntry<Record<string, string>>
+  ): BackendOutputEntry<Record<string, string | Record<string, string>[]>> {
+    const payload = storageOutput.payload;
+    const buckets: {
+      bucketName: string;
+      storageRegion: string;
+    }[] = [];
+    const bucketNames = Object.keys(payload).filter((key) =>
+      key.startsWith('bucketName')
+    );
+    const hasDefaultBucket = bucketNames.includes('bucketName');
+    if (!hasDefaultBucket && bucketNames.length > 1) {
+      throw new AmplifyUserError('NoDefaultBucketError', {
+        message: 'No default bucket set in the Amplify project.',
+        resolution:
+          'Add `isDefault: true` to one of the buckets `defineStorage` in the Amplify project.',
+      });
+    }
```

Comment:
this validation should happen at synthesis time. not here.

Reply:
Thanks. This is the comment I've been waiting for :p

Reply:
Done 1b08c09c20565483c9d856e04dc39407c3aecda6